x-pack/plugins/observability_solution/entity_manager/server/lib/entities/install_entity_definition.ts

```diff
@@ -130,10 +131,12 @@ export async function installBuiltInEntityDefinitions({
       });
     }
 
-    const definition = definitions[0];
+    const definition = installedDefinitions[0];
     // verify current installation
-    if (!definition.state.installed) {
-      logger.debug(`Detected partial installation of definition [${definition.id}], reinstalling`);
+    if (!definition.state.installed || semver.neq(definition.version, builtInDefinition.version)) {
```

Comment:
this should not be necessary but since a failing upgrade would not be fatal (ie shutdown kibana), having logic in the enablement endpoint can help triggering the upgrade without restarting kibana